### DISCUSSION TITLE
Using Unity Catalog with Flink SQL

### DISCUSSION THREAD

**Discussion by rmoff** [unitycatalog/unitycatalog#315] [Ideas]
I'm interested in using Unity with Flink SQL. 

Am I reading the docs correctly to understand that it's not supported yet?

Going the Iceberg route, based on [this](https://docs.databricks.com/en/delta/uniform.html#read-using-the-unity-catalog-iceberg-catalog-endpoint) it seems that I could use the Iceberg REST catalog from Flink to interface with Unity, but Unity Iceberg support is only for reading existing Delta tables

Using Delta, the Delta Flink connector only supports in-memory or Hive catalog, not Unity. 

Am I mistaken on these? thanks!

**Comment by MrPowers** [comment]
On a related note, let's definitely get a Flink integration page added to the docs when this is working!

**Comment by dennyglee** [comment]
Hey @rmoff - this is a great shoutout and we should work together on this I you have the spare cycles.  We're still early and we have.a number of potential approaches, e.g.:
1. There are implementations and PRDs around delta-kernel and delta-kernel-rs for UC support and we could potentially update the delta-flink connector to use delta-kernel
1. As you noted about Iceberg REST Catalog support with UC and ensures it works seamlessly with Iceberg and Delta

I'll let some other folks chime in on preferred approach but either way, it will be important to have UC/Flink integration, eh?!

**Comment by MrPowers** [comment]
Converting this one to a discussion to promote further brainstorming!

**Comment by ognis1205** [comment]
I think this is a really great idea—do you have any updates on it? I’m not trying to rush you or anything, but it’s fantastic to see the ecosystem expanding.
